src/signing.ts

```diff
@@ -77,6 +73,41 @@ export function decode(encoded: EncodedJWS): DagJWS {
     payload: toBase64url(encoded.payload),
     signatures: encoded.signatures.map(decodeSignature),
   }
-  decoded.link = CID.decode(new Uint8Array(encoded.payload))
-  return decoded
+  try {
+    decoded.pld = replaceCIDs(payloadToJSON(encoded.payload)) as Record<string, any>
+    return decoded
+  } catch (e) {
+    try {
+      decoded.link = CID.decode(new Uint8Array(encoded.payload))
+      return decoded
+    } catch (e) {
+      throw new Error('Invalid payload, must be either JSON or CID')
+    }
+  }
+}
+
+function replaceCIDs(data: Record<string, any> | any): Record<string, any> | any {
```

Comment:
might want a `data != null` at the top here or in `isObject` (unless TS is doing something I'm unfamiliar with with typeof)

Reply:
Thanks, added in `isObject`.